### DISCUSSION TITLE
Random order for Combinatorial generation

### DISCUSSION THREAD

**Discussion by andrepiara** [adieyal/sd-dynamic-prompts#323] [Ideas]
Hello! First of all, I want to thank you for all the hard work you’ve put into developing this project.

I have a suggestion for a new feature that I think would be useful. 
Would it be possible to add an option that allows for Combinatorial generations to be generated in a non-sequential order? 
Currently, when generating combinations with two variables (e.g. numbers and letters), the results are displayed in a predetermined sequence (e.g. 1a, 1b, 1c, 2a, 2b, 2c). 
 
It would be great if there was an option to generate these combinations in a random order. 
The reason I’m requesting this feature is that when running large batches of combinations and stopping before completion, having the results generated in a random order would provide more diverse results, otherwise they would look very similar to each other. 
 
 Thank you for considering my request

**Comment by adieyal** [comment]
It's a good idea. It might not be trivial to do though but I'll give it some thought. In the meanwhile, you might consider looking into using random samplers. They don't address your problem entirely but are a way to inject randomness into a combinatorial prompt.

e.g.

```
__day_of__week__ will be {~sunny|rainy|windy|overcast}
```

If you run that in combinatorial mode you will get something like this:
```
Monday will be windy
Tuesday will be sunny
Wednesday will be windy
Thursday will be overcast
Friday will be rainy
Saturday will be overcast
Sunday will be sunny
```

Notice that day of the week is combinatorial, i.e. we get all of the days of the week, but the weather is random. You can read more [here](https://github.com/adieyal/dynamicprompts#samplers)

**Reply by DCNielsen** [reply]
This would be very helpful. Would it be more complex than shuffling the order of the potential generations?

**Comment by adieyal** [comment]
This feature has now been implemented in https://github.com/adieyal/sd-dynamic-prompts/commit/26ce66bbac7a4c5c34d250d45c2327807fad148f
